fedimint-server/src/config/api.rs

```diff
@@ -573,12 +574,13 @@ pub fn server_endpoints() -> Vec<ApiEndpoint<ConfigGenApi>> {
         },
         api_endpoint! {
             "status",
-            async |config: &ConfigGenApi, context, _v: ()| -> ServerStatus {
-                if context.has_auth() {
-                    Ok(config.status().await)
-                } else {
-                    Err(ApiError::unauthorized())
-                }
+            async |config: &ConfigGenApi, _context, _v: ()| -> StatusResponse {
+                let status = config.status().await;
+                assert_ne!(status, ServerStatus::ConsensusRunning);
+                Ok(StatusResponse {
+                    status,
```

Comment:
Maybe this key should be `config_status` instead of just `status`?

Reply:
I think a more long name would something like `{server,general,overall}_status`. It's not specific to config. For instance in future it could say that we are in a `shutdown`/`terminating` state.

But I would like to hear @jkitman  thoughts on that.

Reply:
I would say that status isn't necessarily specific to config gen, just most of the current states are.

For instance we have a `ServerStatus::Upgrade` and `ServerStatus::ConsensusRunning` status.

The only issue with sticking the `Option<ConsensusStatus>` into this struct, is we might eventually need a `Option<UpgradeStatus>` or many other options.  But since this is mostly consumed by the UI, I'll leave it up to @justinmoon to decide whether this seems alright.

Reply:
I think it makes perfect sense to add a new field with a new specific struct when there is more information about a particular status.

Reply:
I'm good with `status`

Reply:
I think @dpc's suggestion make sense, use `server` for `ServerStatus` and `consensus` for `ConsensusStatus`